packages/gatsby/src/commands/develop.ts

```diff
@@ -1,51 +1,74 @@
-const address = require(`address`)
-const cors = require(`cors`)
-const telemetry = require(`gatsby-telemetry`)
-const WorkerPool = require(`../utils/worker/pool`)
-
-const withResolverContext = require(`../schema/context`)
-const sourceNodes = require(`../utils/source-nodes`)
-const createSchemaCustomization = require(`../utils/create-schema-customization`)
-const websocketManager = require(`../utils/websocket-manager`)
-const getSslCert = require(`../utils/get-ssl-cert`)
-const { slash } = require(`gatsby-core-utils`)
-const { initTracer } = require(`../utils/tracer`)
-const apiRunnerNode = require(`../utils/api-runner-node`)
-const db = require(`../db`)
-const detectPortInUseAndPrompt = require(`../utils/detect-port-in-use-and-prompt`)
-const onExit = require(`signal-exit`)
-const queryUtil = require(`../query`)
-const queryWatcher = require(`../query/query-watcher`)
-const requiresWriter = require(`../bootstrap/requires-writer`)
-const {
+import url from "url"
+import fs from "fs"
+import openurl from "better-opn"
+import chokidar from "chokidar"
+
+import { PackageJson } from "gatsby"
+import glob from "glob"
+import express from "express"
+import got from "got"
+import webpack from "webpack"
+import graphqlHTTP from "express-graphql"
+import graphqlPlayground from "graphql-playground-middleware-express"
+import graphiqlExplorer from "gatsby-graphiql-explorer"
+import { formatError } from "graphql"
+
+import webpackConfig from "../utils/webpack.config"
+import bootstrap from "../bootstrap"
+import { store, emitter } from "../redux"
+import { syncStaticDir } from "../utils/get-static-dir"
+import buildHTML from "./build-html"
+import { withBasePath } from "../utils/path"
+import report from "gatsby-cli/lib/reporter"
+import launchEditor from "react-dev-utils/launchEditor"
+import formatWebpackMessages from "react-dev-utils/formatWebpackMessages"
+import chalk from "chalk"
+import address from "address"
+import cors from "cors"
+import telemetry from "gatsby-telemetry"
+import WorkerPool from "../utils/worker/pool"
+import http from "http"
+import https from "https"
+
+import withResolverContext from "../schema/context"
+import sourceNodes from "../utils/source-nodes"
+import createSchemaCustomization from "../utils/create-schema-customization"
+import websocketManager from "../utils/websocket-manager"
+import getSslCert from "../utils/get-ssl-cert"
+import { slash } from "gatsby-core-utils"
+import { initTracer } from "../utils/tracer"
+import apiRunnerNode from "../utils/api-runner-node"
+import db from "../db"
+import detectPortInUseAndPrompt from "../utils/detect-port-in-use-and-prompt"
+import onExit from "signal-exit"
+import queryUtil from "../query"
+import queryWatcher from "../query/query-watcher"
+import requiresWriter from "../bootstrap/requires-writer"
+import {
   reportWebpackWarnings,
   structureWebpackErrors,
-} = require(`../utils/webpack-error-utils`)
+} from "../utils/webpack-error-utils"
+
+type Cert = {
```

Comment:
Do we want to write down guidelines when to use type or interface? I personally use interface for everything except if I need to use union, intersection, ...

Reply:
So I've had so many debates about this. There is no longer a practical difference besides syntax with types and interfaces. I'll switch to interfaces because that's more common in the typescript world. One benefit of interfaces is the practice of prefixing with `I` makes it obvious when importing whats a type and what's code.

e.g., `import { Reporter, IReporter } from 'reporter';`

Reply:
>  I've had so many debates about this

Haha, I can imagine. Honestly, I don't care, I just think something written about our way of using typescript would be great. If an eslint rule exists I would also use it to keep consistency.